packages/dds/tree/src/tableSchema.ts

```diff
@@ -768,31 +759,31 @@ export namespace System_TableSchema {
 						return [];
 					}
 
+					// Ensure the specified columns exists before starting transaction.
+					// Improves user-facing error experience.
+					for (const columnToRemove of columnsToRemove) {
+						this._assertContainsColumn(columnToRemove);
+					}
+
 					const removedColumns: ColumnValueType[] = [];
-					Tree.runTransaction(this, () => {
+					this._applyEditsInBatch(() => {
 						// Note, throwing an error within a transaction will abort the entire transaction.
 						// So if we throw an error here for any column, no columns will be removed.
 						for (const columnToRemove of columnsToRemove) {
-							const removedColumn = this.removeColumn(columnToRemove);
+							const removedColumn = this._removeColumn(columnToRemove);
 							removedColumns.push(removedColumn);
 						}
 					});
 					return removedColumns;
 				}
 			}
 
-			public removeColumn(columnOrId: string | ColumnValueType): ColumnValueType {
+			private _removeColumn(columnOrId: string | ColumnValueType): ColumnValueType {
 				const column = this._getColumn(columnOrId);
```

Comment:
Note for reviewers: I formalized a `get` vs `tryGet` pattern in this PR to unify error handling and such.